exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/traces/SpanReusableDataMarshaler.java

```diff
@@ -20,7 +20,8 @@
  */
 public class SpanReusableDataMarshaler {
 
-  private final Deque<LowAllocationTraceRequestMarshaler> marshalerPool = new ArrayDeque<>();
+  private final SynchronizedQueue<LowAllocationTraceRequestMarshaler> marshalerPool =
```

Comment:
the same construct is also used in https://github.com/open-telemetry/opentelemetry-java/blob/c8da0201abce6e78dfc3bad36fb2fd034da33a4d/exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/logs/LogReusableDataMarshaler.java#L23 and https://github.com/open-telemetry/opentelemetry-java/blob/c8da0201abce6e78dfc3bad36fb2fd034da33a4d/exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/metrics/MetricReusableDataMarshaler.java#L23
did you consider just using a different queue implementation such as `ConcurrentLinkedDeque`?

Reply:
Let's do this.

Also, this change is applicable to [MetricReusableDataMarshaler](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/metrics/MetricReusableDataMarshaler.java) and [LogReusableDataMarshaler](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/common/src/main/java/io/opentelemetry/exporter/internal/otlp/logs/LogReusableDataMarshaler.java) as well.

Reply:
> did you consider just using a different queue implementation such as `ConcurrentLinkedDeque`?

No, I didn't even remember `ConcurrentLinkedDeque` existed, and I'm not familiar with its guarantees and performance characteristics. Might well be a better and lighter option.